x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/AbstractGetResourcesRequest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.action;
+
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.core.action.util.PageParams;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public abstract class AbstractGetResourcesRequest extends ActionRequest {
+
+    public static final String ALLOW_NO_RESOURCES = "allow_no_resources";
```

Comment:
I think this is a bit too abstract to be exposed to end users.  Where we've exposed this in the past it's been in more targeted parameters like `allow_no_jobs`.  So I think each group of classes that extend this class should define their own parameter rather than the abstract class suggesting one.